### ISSUE TITLE
"args": "none" doesn't work

### ISSUE THREAD

**Issue by sarink** [eslint/eslint#11545]
* **ESLint Version:** 5.15.3
* **Node Version:** 11.11.0
* **yarn Version:** 1.13.0

**What parser (default, Babel-ESLint, etc.) are you using?** typescript

**Please show your full configuration:**
(yes, this is the _full_ configuration, `.eslintrc.json`, no extends or overrides of any kind)
```
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": [],
  "plugins": ["react", "react-hooks", "import", "@typescript-eslint"],
  "rules": {
    "import/no-default-export": "warn",
    "no-unused-vars": ["warn", { "args": "none" }],
    "prefer-const": "warn",
    "react/jsx-uses-react": "warn",
    "react/jsx-uses-vars": "warn",
    "react-hooks/rules-of-hooks": "warn",
    "react-hooks/exhaustive-deps": "warn",
    "@typescript-eslint/no-unused-vars": "warn"
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// UserCard.tsx
const useStyles = makeStyles((theme) => {
  return {};
});
```

```bash
yarn run eslint . --ext .ts,.tsx,.js,.jsx
```

**What did you expect to happen?** there should be no warning

**What actually happened? Please include the actual, raw output from ESLint.**
```
/frontend/components/UserCard.tsx
   4:31  warning  'theme' is defined but never used             @typescript-eslint/no-unused-vars
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by platinumazure** [comment]
Hi @sarink, thanks for the issue.

It looks like you've configured the core `no-unused-vars` rule to have `args: "none"`, but you've also enabled the `@typescript-eslint/no-unused-vars` rule with its default settings. According to the error message, the Typescript rule is the one firing the warning.

I would suggest disabling the core rule and seeing if there's a way to configure the Typescript rule with similar configuration options (I assume it's possible because I think that rule is intended to replace the core rule for Typescript code).

**Comment by sarink** [comment]
😱 I didn't even realize that extra rule was in there, haha. Good catch, thanks.
